frontend/src/components/SimpleSelect.tsx

```diff
@@ -56,88 +62,135 @@ const SimpleSelect: React.FC<SimpleSelectProps> = ({
   icon,
   dataTestId,
   toggleProps,
+  previewDescription = true,
+  popperProps,
   ...props
 }) => {
   const [open, setOpen] = React.useState(false);
 
+  const groupedOptionsFlat = React.useMemo(
+    () =>
+      groupedOptions?.reduce<SimpleSelectOption[]>((acc, group) => [...acc, ...group.options], []),
+    [groupedOptions],
+  );
+
   const findOptionForKey = (key: string) =>
-    options?.find((option) => option.key === key) ||
-    groupedOptions
-      ?.reduce<SimpleSelectOption[]>((acc, group) => [...acc, ...group.options], [])
-      .find((o) => o.key === key);
+    options?.find((option) => option.key === key) || groupedOptionsFlat?.find((o) => o.key === key);
 
   const selectedOption = value ? findOptionForKey(value) : undefined;
   const selectedLabel = selectedOption?.label ?? placeholder;
 
+  const totalOptions = React.useMemo(
+    () => [...(options || []), ...(groupedOptionsFlat || [])],
+    [options, groupedOptionsFlat],
+  );
+
+  // If there is only one option, call the onChange function
+  const totalOptionsKey = totalOptions.length === 1 ? totalOptions[0].key : null;
+  React.useEffect(() => {
+    if (totalOptionsKey) {
+      onChange(totalOptionsKey, false);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Maybe the onChange can be memoized here so that we don't have to disable the lint rule here.

Reply:
Do you mean we should do `React.useCallback` on `onChange` here and make it depend on the empty array? I am not sure whether it's working here but I can try. I think disable the linter rule does no harm here, though.

Reply:
I would say this is the type of use case where disabling the lint rule is expected because we do not want the callback to be a dependency. 

Note that while disabling rules is discouraged, there are times when it's necessary. Please always add a comment as to why the rule is disabled when doing so.

Reply:
That's fair. I'll consider this a nitpick then, with the expectation of a possible comment explaining why the lint rule is disabled.